packages/web3-eth-abi/src/utils.ts

```diff
@@ -160,13 +164,9 @@ export const formatParam = (type: string, _param: unknown): unknown => {
 
 	// clone if _param is an object
 	const param = typeof _param === 'object' && !Array.isArray(_param) ? { ..._param } : _param;
-	const paramTypeBytes = /^bytes([0-9]*)$/;
-	const paramTypeBytesArray = /^bytes([0-9]*)\[\]$/;
-	const paramTypeNumber = /^(u?int)([0-9]*)$/;
-	const paramTypeNumberArray = /^(u?int)([0-9]*)\[\]$/;
 
 	// Format BN to string
-	if (param instanceof BigInt) {
+	if (param instanceof BigInt || typeof param === 'bigint') {
```

Comment:
I've passed BigInt(1) and it was not an instance of BigInt. That is why I've added to check the type